pkg/discovery/helper.go

```diff
@@ -109,7 +109,11 @@ func (h *helper) Refresh() error {
 
 	preferredResources, err := h.discoveryClient.ServerPreferredResources()
 	if err != nil {
-		return errors.WithStack(err)
+		if discovery.IsGroupDiscoveryFailedError(err) {
```

Comment:
this isn't a must-change, but since we end up type-asserting the error on the next line anyway, I'd probably instead do this:

```
if discoveryErr, ok := err.(*discovery.ErrGroupDiscoveryFailed); ok {
    // log failed groups, from discoveryErr.Groups
} else {
    return errors.WithStack(err)
}